src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -96,29 +94,51 @@ public static Email parseEmail(String email) throws ParseException {
     }
 
     /**
-     * Parses a {@code String tag} into a {@code Tag}.
+     * Parses a {@code String classGroup} into an {@code ClassGroup}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code classGroup} is invalid.
+     */
+    public static ClassGroup parseClassGroup(String classGroup) throws ParseException {
+        requireNonNull(classGroup);
+        String trimmedClassGroup = classGroup.trim();
+        if (!ClassGroup.isValidClassGroup(trimmedClassGroup)) {
+            throw new ParseException(ClassGroup.MESSAGE_CONSTRAINTS);
+        }
+        return new ClassGroup(trimmedClassGroup);
+    }
+
+    /**
+     * Parses a {@code String telegram} into an {@code Telegram}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code tag} is invalid.
+     * @throws ParseException if the given {@code telegram} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
-        requireNonNull(tag);
-        String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+    public static Telegram parseTelegram(String telegram) throws ParseException {
+        if (telegram.isEmpty()) {
+            return null;
+        }
+        String trimmedTelegram = telegram.trim();
+        if (!Telegram.isValidTelegram(trimmedTelegram)) {
+            throw new ParseException(Telegram.MESSAGE_CONSTRAINTS);
         }
-        return new Tag(trimmedTag);
+        return new Telegram(trimmedTelegram);
     }
 
     /**
-     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     * Parses a {@code String github} into an {@code Github}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code github} is invalid.
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static Github parseGithub(String github) throws ParseException {
+        if (github.isEmpty()) {
+            return null;
+        }
```

Comment:
Should we return Optional<Github> instead? Feels dangerous to return a null when the return type is Github, since the caller would need to know that this value can be null, and won't call methods on this accidentally. Or we could also return `Github.EMPTY` from the suggestion i made on Github.java

Reply:
Returns github.EMPTY here!